app/src/org/commcare/activities/RecoveryActivity.java

```diff
@@ -176,120 +185,111 @@ protected void deliverResult(RecoveryActivity recoveryActivity, Boolean success)
                                 // Try Reinitializing and attemptRecovery again to correct against further issues
                                 CommCareApplication.instance().initializeAppResources(CommCareApplication.instance().getCurrentApp());
                                 attemptRecovery();
+                                loadingIndicator.setVisibility(View.INVISIBLE);
                             } else {
-                                recoveryActivity.displayMessage(StringUtils.getStringRobust(recoveryActivity, R.string.recovery_error));
+                                onRecoveryFailure();

```

Comment:
Now that recovery is, well, a "real" process, I think it would be useful to bubble up more information about this if we can. If nothing else, knowing that recovery is failing due to network issues would be super useful.